### PR TITLE
[AEA-860] libp2p print human readable config

### LINKED ISSUE #1530: Proper logging and exit in p2p startup
**Is your feature request related to a problem? Please describe.**
Currently, when the p2p connection is started with non-matching configs it is not clear from the logs what's wrong. We also do not gracefully terminate the agent.

**Describe the solution you'd like**
Provide multiple configuration scenarios (a: genesis peer with client endpoint; b: genesis peer with no client endpoint; c: non-genesis peer with client end-pont; d: non-genesis peer with no client endpoint; e: behind-nats-peer with client endpoint; f: behind-nats peer with no client endpoint) and properly log which one is selected during startup.

If inconsistent config is provided log this clearly and exit gracefully where necessary.

### PR DESCRIPTION
**Pull request by lrahmani**
## Proposed changes

- Print human readable configuration description of peer following a successful start
- Check that peers public uri ip address matches entry peers ip addresses classes (loopbak-private-public)

Fixes #1530

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 1]
```diff
@@ -536,6 +570,15 @@ def __init__(self, **kwargs):
                     "Local Uri must be set when Public Uri is provided. "
                     "Hint: they are the same for local host/network deployment"
                 )
+            else:
+                # check if node's public host and entry peers hosts are either
+                #  both private or both public
+                if not _ip_all_private_or_all_public(
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/connection.py:576]: avoid nesting with `elif`?

File: packages/fetchai/connections/p2p_libp2p/connection.py [hunk 2]
```diff
@@ -402,7 +437,8 @@ def get_libp2p_node_multiaddrs(self) -> Sequence[MultiAddr]:
             if found:
                 elem = line.strip()
                 if elem != LIST_END:
-                    multiaddrs.append(MultiAddr(elem))
+                    if len(elem) != 0:
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/connection.py:440]: avoid nesting; use `and`?

**Review by DavidMinarsch — APPROVED** [review]
LGTM, but significant coverage gaps are present. Please fill (especially for helpers)

File: packages/fetchai/connections/p2p_libp2p/connection.py (hunk 1 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/connection.py]: removed the nesting

**Pull request merged**